sdk/python/feast/go_server.py

```diff
@@ -0,0 +1,246 @@
+                determines which features will be retrieved. Feature references should
+                be of the form "feature_view:feature".
+            entity_rows: A list of dictionaries where each key-value pair is an
+                entity-name or entity-value pair.
+            full_feature_names: Whether feature names should be returned with feature
+                view names as prefixes, changing them from the format "feature" to
+                "feature_view__feature".
+
+        Returns:
+            An OnlineResponse containing the feature data.
+
+        Raises:
+            InvalidFeaturesParameterType: If features is not a list or a feature service.
+            FeatureNameCollisionError: If a feature reference is specified multiple times.
+            ValueError: If some other error occurs.
+        """
+        if not self.grpc_server_started:
+            self._connect()
+
+        request = GetOnlineFeaturesRequest(full_feature_names=full_feature_names)
+        if isinstance(features, FeatureService):
+            request.feature_service = features.name
+        elif isinstance(features, list):
+            request.features.val.extend(features)
+        else:
+            raise InvalidFeaturesParameterType(features)
+
+        for key, values in entities.items():
+            request.entities[key].val.extend(python_values_to_proto_values(values))
+
+        try:
+            response = self.client.GetOnlineFeatures(request=request)
+        except grpc.RpcError as rpc_error:
+            # If the server became unavailable, it could mean that the subprocess died
+            # or fell into a bad state, so the resolution is to establish a new Go
+            # subprocess and set up a new connection with it.
+
+            # Socket might not have closed if this is a grpc problem.
+            self.stop()
+            if rpc_error.code() == grpc.StatusCode.UNAVAILABLE:
+                self._connect()
+                # Retry request with the new Go subprocess
+                response = self.client.GetOnlineFeatures(request=request)
+            else:
+                error_message = rpc_error.details()
+                if error_message.startswith(FeatureNameCollisionError.__name__):
+                    parsed_error_message = error_message.split(": ")[1].split("; ")
+                    collided_feature_refs = parsed_error_message[0].split(", ")
+                    full_feature_names = parsed_error_message[1] == "true"
+                    raise FeatureNameCollisionError(
+                        collided_feature_refs, full_feature_names
+                    )
+                elif error_message.startswith(ValueError.__name__):
+                    parsed_error_message = error_message.split(": ")[1]
+                    raise ValueError(parsed_error_message)
+                else:
+                    raise
+
+        return OnlineResponse(response)
+
+    def _connect(self):
```

Comment:
A bit of a strange name, given that we are not just connecting but also spawning a process.

Reply:
What about _start_go_server?

Reply:
sgtm